### DISCUSSION TITLE
How to connect Bonsai with USB3.0 video camera

### DISCUSSION THREAD

**Discussion by RenOdun** [bonsai-rx/bonsai#1654] [Q&A] [answered]
Hi everyone,
Our lab bought some industrial camera for observing animal behavior with high fps. It seems that bonsai can only recognize plug and play USB2.0 camera derectly. Is there any tutorial or packages for connecting USB3.0 video camera?
I found a website, but the link is disabled. https://groups.google.com/g/bonsai-users/c/7jdh86LKkIw

**Comment by bruno-f-cruz** [comment]
Hey! Bonsai is agnostic to the USB protocol version. The generic backends (directshow and opencv) are what determine the compatibility of the operators (videocapturedevice and cameracapture, respectively). If your camera is not compatible you should ask the company to provide you drivers. 
Out of curiosity, what is the camera model?

**Reply by RenOdun** [reply]
Thanks for such a rapid response!
Those camera is Hikvision MV-CE013-80UM, and we used their own software till now. 
https://www.hikrobotics.com/cn
When recording the pictures, the cameras always lose frames while the exposure signal were recored. Therefore we seek other software to record the readout signals along with behavior task signals.
Actually, the company has provided some secondly-development samples, but I don`t know what and how to do for connection.
![微信截图_20240108211752](https://github.com/bonsai-rx/bonsai/assets/73214100/b871a76e-e5fa-4776-9844-2379822a9600)

**Reply by RenOdun** [reply]
The disabled linkages is:
https://bitbucket.org/horizongir/bonsai/src/c5d4e340bdea0d033e4962cbe8dff5159f639209/Bonsai.Vision/CameraCapture.cs?at=default&fileviewer=file-view-default
https://bitbucket.org/horizongir/bonsai/wiki/PackageTutorial

**Reply by bruno-f-cruz** [reply]
These cameras probably require their communication via their SDK unfortunately (the website you sent seems to be down?). That c# example might be a good place to start.

**Reply by RenOdun** [reply]
Those websides were provided in https://groups.google.com/g/bonsai-users/c/7jdh86LKkIw for a custom Bonsai source, but they are unaccessible now. 
Do you have some examples for programing the bonsai source in C#? Or could you enumerate some other brand camera supporeted in Bonsai?

**Reply by bruno-f-cruz** [reply, marked as answer]
The updated instructions to build your own package are [here](https://bonsai-rx.org/docs/articles/create-package.html).
Current packages for computer vision cameras include Pylon and Spinnaker.

**Reply by RenOdun** [reply]
I found directshow installers in the company files, and Bonsai recognize camera immediately! 
![微信截图_20240108214318](https://github.com/bonsai-rx/bonsai/assets/73214100/43c25dd0-cac1-4a3b-b67f-6fddc27ea39b)
HaHa, thanks for answering!
